xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-maven/src/main/java/org/xwiki/extension/maven/internal/MavenExtensionScanner.java

```diff
@@ -318,7 +318,9 @@ public void guess(Map<String, DefaultCoreExtension> extensions, Collection<URL>
                 if (coreExtension == null) {
                     String dependencyFileName = dependency.getArtifactId() + '-' + dependency.getVersion();
                     if (dependency.getClassifier() != null) {
-                        dependencyFileName += '-' + dependency.getClassifier();
+                        StringBuilder stringBuilder = new StringBuilder(dependency.getClassifier());
```

Comment:
This code does not get the same result that the previous one. Also StringBuilder should be used as earlier to be fully accurate since there is concatenation in previous line too.